src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Compiler/CachePoolPrunerPassTest.php

```diff
@@ -44,27 +44,17 @@ public function testCompilerPassReplacesCommandArgument()
 
     public function testCompilePassIsIgnoredIfCommandDoesNotExist()
     {
-        $container = $this
-            ->getMockBuilder(ContainerBuilder::class)
-            ->setMethods(array('hasDefinition', 'getDefinition', 'findTaggedServiceIds'))
-            ->getMock();
-
-        $container
-            ->expects($this->atLeastOnce())
-            ->method('hasDefinition')
-            ->with(CachePoolPruneCommand::class)
-            ->will($this->returnValue(false));
-
-        $container
-            ->expects($this->never())
-            ->method('getDefinition');
+        $container = new ContainerBuilder();
 
-        $container
-            ->expects($this->never())
-            ->method('findTaggedServiceIds');
+        $definitionsBefore = count($container->getDefinitions());
+        $aliasesBefore = count($container->getAliases());
 
         $pass = new CachePoolPrunerPass();
         $pass->process($container);
+
+        // the container is untouched (i.e. no new definitions or aliases)
```

Comment:
this still allows altering the existing definitions (well, except that you use an empty builder in this test)